data-analysis/template.yaml

```diff
@@ -17,6 +17,11 @@ Conditions:
   LockCrossAccountPermissionsToLambdaRoles:
     !Not [!Equals [!Ref Environment, dev]]
   IsProduction: !Equals [!Ref Environment, 'production']
+  NotTestEnvironment:
```

Comment:
I feel like I'd prefer this condition the other way around. I.e `TestEnvironment` which we've used elsewhere in the codebase. You could either wrap this in a !Not function, or just add the other envs in the !O